site/docs/v1/tech/apis/_tenant-request-body.adoc

```diff
@@ -415,6 +435,20 @@ Indicates that users without a verified email address will be permanently delete
 [field]#tenant.userDeletePolicy.unverified.numberOfDaysToRetain# [type]#[Integer]# [optional]#Optional# [since]#Available since 1.13.0#::
 The number of days from creation users will be retained before being deleted for not completing email verification. This field is required when [field]#tenant.userDeletePolicy.unverified.enabled# is set to `true`.  Value must be greater than 0.
 
+[field]#tenant.usernameConfiguration.unique.enabled# [type]#[Boolean]# [optional]#Optional# [default]#defaults to `false`# [since]#Available since 1.27.0#::
+When `true`, FusionAuth will handle username collisions by generating a random suffix.
++
+:premium_feature: unique usernames
+include::../shared/_premium-edition-blurb-api.adoc[]
++
+:premium_feature!:
+
+[field]#tenant.usernameConfiguration.unique.numberOfDigits# [type]#[Integer]# [optional]#Optional# [default]#defaults to `5`# [since]#Available since 1.27.0#::
+The maximum number of digits to use when building a unique suffix for a username. A number will be randomly selected and may be 1 or more digits up to this configured value. Value can be `>=3` and `<=10`.
+
+[field]#tenant.usernameConfiguration.unique.separator# [type]#[String]# [optional]#Optional# [default]#defaults to `&#35;`# [since]#Available since 1.27.0#::
```

Comment:
```suggestion
[field]#tenant.usernameConfiguration.unique.separator# [type]#[String]# [optional]#Optional# [default]#defaults to the `&#35;` character# [since]#Available since 1.27.0#::
```